ui/src/gba_display.rs

```diff
@@ -78,12 +78,13 @@ impl View for GbaDisplay {
             }
         });
 
-        if let Ok(gba) = self.gba.lock() {
-            gba.ppu.render();
-            for row in 0..DISPLAY_HEIGHT {
-                for col in 0..DISPLAY_WIDTH {
-                    self.image[(col, row)] = GbaColor(gba.lcd.borrow()[(col, row)]).into();
-                }
+        let gba = self.gba.lock().unwrap();
+        let gba_lcd = gba.lcd.lock().unwrap();
+
+        gba.ppu.render();
```

Comment:
```suggestion
        let gba = self.gba.lock().unwrap();
         gba.ppu.render();
         
        let gba_lcd = gba.lcd.lock().unwrap();
```
I suggest to change the order of these two instructions. I try to run the code but the GUI stucks in an eternal loop. I think that the reason is because `ppu.render()` tries to get a lock, but the UI alredy has it.

Reply:
Mh ok so for now, as I understood, UI updates every window every step, right?

With my changes since we spawn a thread that continuously lock the `gba` the other UI tools can't updated themselves, correct?

I can fix this specific case but at the moment I think we're bit far to have a correct "play loop" (cpu executes, irqs get triggered, lcd refreshed, etc...) because we miss irqs and timings and many other things.

If you're ok with it (@guerinoni, @gallottino), for now I would treat this "play" button as a fast "go to the first failure", and we come back to it once we implement more stuff in the emulator.